src/VSCodeExtension/src/quickPickJob.ts

```diff
@@ -0,0 +1,255 @@
+import * as vscode from "vscode";
+import * as excludedTargets from "./utils/excludedTargets.json";
+import {workspaceInfo} from "./commands";
+import { QuantumJobClient } from "@azure/quantum-jobs";
+type target = {
+    id: string;
+    currentAvailability: string;
+    averageQueueTime: number;
+    statusPage: string;
+  };
+
+type jobMetaData = {provider:string, target:string, jobName:string, programArguments:string};
+
+let updateWorkspace = false;
+let quickPickIsHidden = false;
+
+export async function getJobInfoFromUser(context:vscode.ExtensionContext, quantumJobClient:QuantumJobClient, workspaceInfo:workspaceInfo, update=false){
+
+    updateWorkspace = update;
+    let provider:string;
+    let target:string;
+    let jobName:string;
+    let programArguments:string;
+
+
+    let providersAndTargets:any;
+
+    return new Promise<jobMetaData>( async (resolve, reject)=>{
+        const quickPick = vscode.window.createQuickPick();
+        quickPickIsHidden = false;
+        const inputBox = vscode.window.createInputBox();
+        inputBox.buttons = [vscode.QuickInputButtons.Back];
+        // if user has to select workspace, total steps will be 7, otherwise 4
+        inputBox.totalSteps = updateWorkspace?7:4;
+        quickPick.totalSteps = updateWorkspace?7:4;
+        inputBox.ignoreFocusOut = true;
+        quickPick.ignoreFocusOut = true;
+
+
+        providersAndTargets = await setupProviderQuickPick(quickPick, context, quantumJobClient, workspaceInfo, inputBox);
+        if(providersAndTargets===undefined){
+            inputBox.dispose();
+            quickPick.dispose();
+            reject();
+        }
+
+
+         inputBox.onDidAccept(async () => {
+          // final step finished, therefore submit job
+            if((!updateWorkspace && inputBox.step===4)||(updateWorkspace && inputBox.step===7)){
+                programArguments = inputBox.value;
+                inputBox.dispose();
+                quickPick.dispose();
+                resolve({jobName:jobName,provider:provider, target:target,programArguments:programArguments});
+            }
+          // user input job name so prompt program arguments input
+            if ((!updateWorkspace && inputBox.step===3)||(updateWorkspace && inputBox.step===6)){
```

Comment:
Use a numeric enum instead. It will be much easier to maintain if we know what the steps are. Also, there might be steps that could be inserted or removed. You can also use repeated enum values for your 7 or 4 counts in lines 34 and 35.